src/openapi/streaming/subscription.ts

```diff
@@ -934,6 +986,10 @@ class Subscription {
                 // We are about to unsubscribe by tag, so no need to do anything
                 return;
 
+            case this.STATE_PUBLISHER_DOWN:

```

Comment:
What about including this one in the first case?

Reply:
took inspiration from `STATE_READY_FOR_UNSUBSCRIBE_BY_TAG` 
that will work but i took it out for more readability, also 
```
case this.STATE_UNSUBSCRIBED:
            case this.STATE_UNSUBSCRIBE_REQUESTED:
```
looks more concerned related to unsubscribe, so thought better to keep it separate.